tests/extractors/licence_test.py

```diff
@@ -0,0 +1,64 @@
+import json
+import logging
+import re
+from concurrent.futures import Executor
+from pathlib import Path
+
+import pytest
+from tldextract import TLDExtract
+
+from app.models import Input, StarCase
+from app.splash_models import SplashResponse
+from core.website_manager import WebsiteData
+from features.licence import Licence, LicenceExtractor
+
+
+@pytest.mark.asyncio
+async def test_licence_extractor(executor: Executor):
+    with open(Path(__file__).parent.parent / "resources" / "splash-response-google.json", "r") as f:
+        splash_response = SplashResponse.parse_obj(json.load(f))
+    # we dont want to count "CC BY-SA" as "CC BY"!
+    splash_response.html = "Lorem ipsum Creative Commons Zero ... CC BY-SA ... CC BY ... CC0 ..."
+    site = await WebsiteData.from_input(
+        input=Input(url="https://www.google.com", splash_response=splash_response),
+        tld_extractor=TLDExtract(cache_dir=None),
+        logger=logging.getLogger(),
+    )
+
+    extractor = LicenceExtractor()
+    await extractor.setup()
+    stars, explanation, extra = await extractor.extract(site=site, executor=executor)
+    assert extra.guess is Licence.CC0
+    assert extra.total == 4
+    assert len(extra.counts) == 3
+    assert set(extra.counts.keys()) == {Licence.CC0, Licence.CC_BY, Licence.CC_BY_SA}
+    assert stars == StarCase.FIVE
+    assert isinstance(explanation, str), "Explanation should be a string"
+
+
+def test_regex():
+    html = "Lorem ipsum Creative Commons Zero ... CC BY-SA ... CC BY ... CC0 ..."
+    pattern = "(?P<foo>(CC BY))|(CC BY-SA)|(Creative Commons Zero|CC0)"
+    pattern = re.compile(pattern)
+    print(pattern.flags, pattern.groups, pattern.groupindex)
+    for i in pattern.findall(html):
+        print(i)
+
+    # print(pattern.findall(html))
+
+
+# fixme: remove or activate. This reads through a set of splash responses and analyzes them to get an idea how many
```

Comment:
Instead of commenting, either skip it with pytest - so it appears in pytest and refactoring - or delete it.

Reply:
Yes, maybe we can have a quick chat about that today. It also somewhat touches the urls.txt topic.

Reply:
removed it